dev/src/transaction.ts

```diff
@@ -552,3 +561,14 @@ function validateReadOptions(
     }
   }
 }
+
+function isRetryableTransactionError(error: Error): boolean {
+  if (error instanceof GoogleError || 'code' in error) {
```

Comment:
Yes, that is JavaScript functionality (see https://masteringjs.io/tutorials/fundamentals/hasownproperty for a comparison against `Object.hasOwnProperty`). I am doing it here because I am afraid that some of our dependencies might throw a GoogleError that is not actually a GoogleError by providing just a code and a message field.